src/favs/favs.service.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  forwardRef,
+  Inject,
+  Injectable,
+  NotFoundException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { AlbumService } from 'src/album/album.service';
+import { ArtistService } from 'src/artist/artist.service';
+import { TrackService } from 'src/track/track.service';
+import { FavoritesResponse } from './interfaces/favs.interface';
+
+@Injectable()
+export class FavsService {
+  constructor(
+    @Inject(forwardRef(() => ArtistService))
+    private readonly artistService: ArtistService,
+    @Inject(forwardRef(() => AlbumService))
+    private readonly albumService: AlbumService,
+    @Inject(forwardRef(() => TrackService))
+    private readonly trackService: TrackService,
+  ) {}
+
+  private favs: FavoritesResponse = {
+    artists: [],
+    albums: [],
+    tracks: [],
+  };
```

Comment:
You store liked tracks/albums/artists inside favorites service. In general, the logic is spead between 2 services and you are responsible for maintaining it.

Another approach would to let `Album`, `Track` and `Artist` objects to store the information whether it is liked or unliked. This way, logic would reside in rich domain entities.
https://medium.com/@inzuael/anemic-domain-model-vs-rich-domain-model-78752b46098f